transport/src/test/java/io/camunda/zeebe/transport/stream/impl/StreamIntegrationTest.java

```diff
@@ -318,6 +338,35 @@ void shouldRemoveAllStreamsOnClientShutdown() {
             awaitStreamOnServer(streamType, server2, stream -> assertThat(stream).isEmpty());
           });
     }
+
+    @Test
+    void shouldAddStreamAgainOnServerRestart() {
+      // given
+      final var streamType = BufferUtil.wrapString("foo");
+      final var properties = new TestSerializableData();
+      final var streamId =
+          clientStreamer
+              .add(streamType, properties, p -> CompletableFuture.completedFuture(null))
+              .join();
+      awaitStreamAdded(streamType, streamId, server1, server2);
+      // since we can't really restart the server (at least until we can restart AtomixCluster
+      // instances), we'll just simulate it by removing it; this means the stream is still
+      // registered server side, so we only wait for it to be removed client side
+      client.streamService.onServerRemoved(server1.memberId());
```

Comment:
:wrench: This would not have caught the bug, right? Can't we simulate restart by closing and creating a new server with same id? 
```
      server1.close();
      client.streamService.onServerRemoved(server1.memberId());
      awaitStreamOnClient(
          streamId,
          maybeStream -> {
            assertThat(maybeStream).isPresent();
            assertThat(maybeStream.get().isConnected(server1.memberId())).isFalse();
          });

      // when - reuse memberId of server1 to simulate restart
      final var restartedServer1 = new TestServer(createClusterNode(clusterNodes.get(0), clusterNodes));
      restartedServer1.start();
      client.streamService.onServerJoined(restartedServer1.memberId());
```

Reply:
You're right, I was being lazy and avoiding more refactoring - we can of course recreate the server entirely :+1:

Reply:
Though I did test without the fix and the test failed, so it would catch something at least ;)